src/app/shared/filter/date-range.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { FilterModel } from '../../ngrx/model';
+import { castleFilter } from '../../ngrx/actions/castle.action.creator';
+
+@Component({
+  selector: 'metrics-date-range',
+  template: `
+    <div>
+      <span>From:</span>
+      <span>
+        <prx-datepicker [date]="filter.beginDate" UTC="true" (dateChange)="onBeginDateChange($event)"></prx-datepicker>
+        <prx-timepicker [date]="filter.beginDate" UTC="true" (timeChange)="onBeginDateChange($event)"></prx-timepicker>
+      </span>
+    </div>
+    <div>
+      <span>Through:</span>
+      <span>
+        <prx-datepicker [date]="filter.endDate" UTC="true" (dateChange)="onEndDateChange($event)"></prx-datepicker>
```

Comment:
putting here as a reminder to discuss -- should we make a styleguide ticket to optionally disable dates past today? not allowing selection of dates later than today here would match the disabling of the `NEXT` button for ranges past today

Reply:
opened this ticket: https://github.com/PRX/metrics.prx.org/issues/31